snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,152 @@
+        return Param(
+            "uuid_column" + suffix, ColumnMatch(None, self.__uuid_column_match)
+        )
+
+    def __init__(self, uuid_columns: Set[str]) -> None:
+        self.__unique_uuid_columns = set(uuid_columns)
+        self.__uuid_column_match = Or(
+            [String(u_col) for u_col in self.__unique_uuid_columns]
+        )
+        self.uuid_in_condition = FunctionCallMatch(
+            Or((String(ConditionFunctions.IN), String(ConditionFunctions.NOT_IN))),
+            (
+                Or((self.uuid_column_pattern(), self.formatted_uuid_pattern())),
+                Param("params", FunctionCallMatch(String("tuple"), None)),
+            ),
+        )
+        self.uuid_condition = FunctionCallMatch(
+            Or([String(op) for op in FUNCTION_TO_OPERATOR]),
+            (
+                Or(
+                    (
+                        Param("literal_0", LiteralMatch(AnyOptionalString())),
+                        self.uuid_column_pattern("_0"),
+                        self.formatted_uuid_pattern("_0"),
+                    )
+                ),
+                Or(
+                    (
+                        Param("literal_1", LiteralMatch(AnyOptionalString())),
+                        self.uuid_column_pattern("_1"),
+                        self.formatted_uuid_pattern("_1"),
+                    )
+                ),
+            ),
+        )
+
+    def parse_uuid(self, lit: Expression) -> Expression:
+        if not isinstance(lit, Literal):
+            return lit
+
+        try:
+            parsed = uuid.UUID(str(lit.value))
+            return Literal(lit.alias, str(parsed))
+        except Exception:
+            return lit
+
+    def process_condition(self, exp: Expression) -> Expression:
+        if not isinstance(exp, FunctionCall):
+            return exp
+
+        result = self.uuid_in_condition.match(exp)
+        if result is not None:
+            new_params: List[Expression] = []
+            if result.contains("formatted_uuid_column"):
+                column = result.expression("formatted_uuid_column")
+                assert isinstance(column, Column)
+                new_params.append(Column(None, column.table_name, column.column_name))
+            else:
+                column = result.expression("uuid_column")
+                assert isinstance(column, Column)
+                new_params.append(column)
+
+            params_fn = result.expression("params")
+            assert isinstance(params_fn, FunctionCall)
+            new_fn_params = []
+            for param in params_fn.parameters:
+                new_fn_params.append(self.parse_uuid(param))
+
+            new_params.append(
+                FunctionCall(
+                    params_fn.alias, params_fn.function_name, tuple(new_fn_params)
+                )
+            )
+
+            return binary_condition(
+                exp.alias, exp.function_name, new_params[0], new_params[1]
```

Comment:
new_params is always a sequence of two elements that you are exploding here anyway. It would be more readable if you just gave a name to each and avoid the list. Then by reading this statement you would easily know what is new_param[0] and what is new_param[1] without trying to understand why this is a list.